types/tasks.ts

```diff
@@ -24,3 +24,27 @@ export type Patch = RadiclePatch & {
 export type RadicleTask = RadicleIssue | RadiclePatch
 
 export type Task = Issue | Patch
+
+/**
+ * Represents the text properties of a task to be highlighted.
+ *
+ * Each string array represents a string, split in segments so that odd indexes are
+ * highlighted while even indexes are not.
+ *
+ * @example
+ * ```ts
+ * // The **quick** brown fox jumps over the lazy **dog**.
+ * ['The ', 'quick', ' brown fox jumps over the lazy ', 'dog', '.']
+ * ```
+ *
+ * @example
+ * ```ts
+ * // **The** quick brown fox jumps over the lazy **dog.**
+ * ['', 'The', ' quick brown fox jumps over the lazy ', 'dog.']
+ * ```
+ */
+export interface TaskHighlights {
```

Comment:
Tried a couple of things before ending up with this.

---
One was similar but with indexes for the highlighted segments and in combination with the original string you could compute the highlights.
 (e.g. `**fo**o b**a**r` would become `[[0, 1], [5,5]]`)

This felt like an extra step of complexity compared to the current solution.

---

Another thing I tried was passing the query to each card and then having the card perform the matching and highlighting internally (the filtering was still done on the tasks store).

That felt like a lot of logic for the card, and also not very flexible (e.g. I doubt it would work if we added fuzzy search)